src/dmd/dsymbolsem.d

```diff
@@ -805,9 +806,11 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             // so mark the scope as ctfe if required
             bool needctfe = (dsym.storage_class & (STC.manifest | STC.static_)) != 0;
             if (needctfe)
+            {
+                sc.flags |= SCOPE.condition;
```

Comment:
@benjones This breaks the error for `is` *Identifier* forms used outside of static if/assert:
https://issues.dlang.org/show_bug.cgi?id=23827

I tried making a fix but it doesn't pass the test here.

Reply:
I saw your new PR.  It's been so long, I don't remember how I came to think this was the right fix and probably won't have time to look closely in the next couple of weeks.